clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -434,8 +434,9 @@ private static int parseAcks(String acksString) {
             int serializedSize = Records.LOG_OVERHEAD + Record.recordSize(serializedKey, serializedValue);
```

Comment:
Since we are adding timestamp field to ProducerRecord and RecordMetadata, I think we should add a comment to KafkaProducer.send() description what happens to timestamp in ProducerRecord (e.g., if it is of type CreateType, then the value would be set to local time on producer) and what value they will get back in the callback.